design-documents/graph-ql/coverage/returns.md

```diff
@@ -0,0 +1,350 @@
+## Configuration 
+
+The following settings should be accessible via `storeConfig` query:
+- Returns functionality status on the storefront: enabled/disabled
+
+Scenarios which may need these settings include:
+- Rendering of the returns section in the customer account
+
+```graphql
+{
+  storeConfig {
+    sales_magento_rma_enabled
+  }
+}
+```
+
+## Use cases
+
+### View return list with pagination in customer account
+
+```graphql
+{
+  customer {
+    returns(pageSize: 10, currentPage: 2) {
+      items {
+        id
+        creation_date
+        customer_name
+        status
+      }
+      page_info {
+        current_page
+        page_size
+        total_pages
+      }
+      total_count
+    }
+  }
+}
+```
+
+### View return list with pagination in order details
+
+```graphql
+{
+  customer {
+    orders(filter: {number: {eq: "00000008"}}) {
+      items {
+        returns(pageSize: 10, currentPage: 1) {
+          items {
+            id
+            creation_date
+            customer_name
+            status
+          }
+          page_info {
+            current_page
+            page_size
+            total_pages
+          }
+          total_count
+        }
+      }
+    }
+  }
+}
+```
+
+### View return details
+
+```graphql
+{
+  customer {
+    return(id: "0000003") {
+      id
+      order_id
+      creation_date
+      customer_email
+      customer_name
+      status
+      comments {
+        id
+        text
+        created_at
+        created_by
+      }
+      items {
+        id
+        product {
+          sku
+          name
+        }
+        custom_attributes {
+          id
+          label
+          value
+        }
+        request_quantity
+        quantity
+        status
+      }
+      shipping {
+        tracking {
+          id
+          carier {
+            label
+          }
+          tracking_number
+        }
+        address {
+          contact_name
+          street
+          city
+          region {
+            code
+          }
+          country {
+            full_name_english
+          }
+          postcode
+          telephone
+        }
+      }
+    }
+  }
+}
+```
+
+### Create a return
+
+#### Determine whether any order items are eligible for return
```

Comment:
re: Anton's comment https://github.com/magento/architecture/pull/397/files#r458432579

> Why separate list instead of marking order items as eligible for return?

For new UIs that aren't Luma, they may want to visually distinguish items that can be returned, while still on the order details page.

It would be a lot simpler for the UI if this data was located in the order line items, rather than in a separate list requires nested loops or client-side normalization.

Reply:
@DrewML We can add it to order line items for such use cases. But isn't it simpler for current UI to have a separate list? See my other reply to the Anton's comment.

Reply:
Added support for:

```
{
  customer {
    orders(filter: {number: {eq: "00000008"}}) {
      items {
        items {
          id
          eligible_for_return
        }
      }
    }
  }
}
```